skt/runner.py

```diff
@@ -137,14 +136,14 @@ def __init__(self, jobtemplate, jobowner=None, blacklist=None):
         # Set up the default, allowing for overrides with each run
         self.max_aborted = 3
 
-        # determines if termination cleanup was done and all jobs terminated
-        self.cleanup_done = False
-
         # if True, keep waived tests hidden for this run
         self.waiving = None
 
-        # retcode used here only to decided if cleanup handler should be run
-        self.retcode = None
+        # the actual retcode to return is stored here
+        self.retcode = SKT_ERROR
+
+        # if True, cleanup handler was run for this runner
+        self.cleanup_done = False
```

Comment:
Since the runner has nothing to do with cleanup now, please remove `cleanup_done` from it, and move it to the executable completely.

Reply:
Any thoughts on where to put that variable so it can be accessed from the test without making it a global variable?

Reply:
I would instead base the test on something else. Like the update of the configuration file.